server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSSysSetOptionsCheckUtility.java

```diff
@@ -0,0 +1,1537 @@
+        checkHasMandatoryOptions(ctx.DOCTEMPLATE(), ctx, "DOCTEMPLATE");
+
+        checkHasExactlyOneOption("COPY or NEWCOPY", ctx, ctx.COPY(), ctx.NEWCOPY());
+    }
+
+    private void checkDsname(CICSParser.Cics_set_dsnameContext ctx) {
+        checkHasMandatoryOptions(ctx.DSNAME(), ctx, "DSNAME");
+
+        checkMutuallyExclusiveOptions("ACTION, REMOVE, RECOVERED, RESETLOCKS or RETRY", ctx.ACTION(), ctx.REMOVE(), ctx.RECOVERED(), ctx.RESETLOCKS(), ctx.RETRY());
+        checkMutuallyExclusiveOptions("AVAILABILITY, AVAILABLE, RREPL or UNAVAILABLE", ctx.AVAILABILITY(), ctx.AVAILABLE(), ctx.RREPL(), ctx.UNAVAILABLE());
+        checkMutuallyExclusiveOptions("QUESCESTATE, QUIESCED, IMMQUIESCED or UNQUIESCED", ctx.QUESCESTATE(), ctx.QUIESCED(), ctx.IMMQUIESCED(), ctx.UNQUIESCED());
+        checkMutuallyExclusiveOptions("WAIT, BUSY or NOWAIT", ctx.WAIT(), ctx.BUSY(), ctx.NOWAIT());
+        checkMutuallyExclusiveOptions("UOWACTION, BACKOUT, COMMIT or FORCE", ctx.UOWACTION(), ctx.BACKOUT(), ctx.COMMIT(), ctx.FORCE());
+    }
+
+    private void checkDumpds(CICSParser.Cics_set_dumpdsContext ctx) {
+        checkHasMandatoryOptions(ctx.DUMPDS(), ctx, "DUMPDS");
+
+        checkMutuallyExclusiveOptions("OPENSTATUS, CLOSED, OPEN or SWITCH", ctx.OPENSTATUS(), ctx.CLOSED(), ctx.OPEN(), ctx.SWITCH());
+        checkMutuallyExclusiveOptions("SWITCHSTATUS, NOSWITCH, SWITCHNEXT or SWITCHALL", ctx.SWITCHSTATUS(), ctx.NOSWITCH(), ctx.SWITCHNEXT(), ctx.SWITCHALL());
+    }
+
+    private void checkEnqmodel(CICSParser.Cics_set_enqmodelContext ctx) {
+        checkHasMandatoryOptions(ctx.ENQMODEL(), ctx, "ENQMODEL");
+
+        checkHasExactlyOneOption("STATUS, ENABLED or DISABLED", ctx, ctx.STATUS(), ctx.ENABLED(), ctx.DISABLED());
+    }
+
+    private void checkEpadapter(CICSParser.Cics_set_epadapterContext ctx) {
+        checkHasMandatoryOptions(ctx.EPADAPTER(), ctx, "EPADAPTER");
+
+        checkMutuallyExclusiveOptions("ENABLESTATUS, ENABLED or DISABLED", ctx.ENABLESTATUS(), ctx.ENABLED(), ctx.DISABLED());
+    }
+
+    private void checkEpadapterset(CICSParser.Cics_set_epadaptersetContext ctx) {
+        checkHasMandatoryOptions(ctx.EPADAPTERSET(), ctx, "EPADAPTERSET");
+    }
+
+    private void checkEventbinding(CICSParser.Cics_set_eventbindingContext ctx) {
+        checkHasMandatoryOptions(ctx.EVENTBINDING(), ctx, "EVENTBINDING");
+
+        checkMutuallyExclusiveOptions("ENABLESTATUS, ENABLED or DISABLED", ctx.ENABLESTATUS(), ctx.ENABLED(), ctx.DISABLED());
+    }
+
+    private void checkEventprocess(CICSParser.Cics_set_eventprocessContext ctx) {
+        checkHasMandatoryOptions(ctx.EVENTPROCESS(), ctx, "EVENTPROCESS");
+
+        checkHasExactlyOneOption("EPSTATUS, STARTED, DRAIN or STOPPED", ctx, ctx.EPSTATUS(), ctx.STARTED(), ctx.DRAIN(), ctx.STOPPED());
+    }
+
+    private void checkFile(CICSParser.Cics_set_fileContext ctx) {
+        if (!ctx.DATASET().isEmpty())
+            checkHasMandatoryOptions(ctx.FILE(), ctx, "FILE");
```

Comment:
this needs the exactly one check - using both options at the same time is not allowed.